metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -545,9 +550,17 @@ object ImplementationProvider {
     def fromClassSignature(
         classSig: ClassSignature
     ): Seq[(String, ClassLocation)] = {
-      classSig.parents.collect { case t: TypeRef =>
-        t.symbol -> ClassLocation(symbol, filePath.map(_.toNIO))
+      // support `&`-types here?
```

Comment:
Should we have a follow up issue here?

Reply:
Added the support for intersection types in https://github.com/scalameta/metals/pull/7170/commits/1da9dd788df99dbdddf3775d794059e10bf76f2f, otherwise it's only rebase.